compiler/src/lowering/mappers/functions.rs

```diff
@@ -1 +1,127 @@
+use crate::lowering::mappers::expressions;
+use fe_analyzer::Context;
+use fe_parser::ast as fe;
+use fe_parser::node::Node;
 
+pub fn func_def(context: &Context, def: Node<fe::ContractStmt>) -> Node<fe::ContractStmt> {
+    if let fe::ContractStmt::FuncDef {
+        qual,
+        name,
+        args,
+        return_type,
+        body,
+    } = def.kind
+    {
+        let lowered_body = multiple_stmts(context, body);
+
+        let lowered_kind = fe::ContractStmt::FuncDef {
+            qual,
+            name,
+            args,
+            return_type,
+            body: lowered_body,
+        };
+
+        return Node::new(lowered_kind, def.span);
+    }
+
+    unreachable!()
+}
+
+fn func_stmt(context: &Context, stmt: Node<fe::FuncStmt>) -> Vec<Node<fe::FuncStmt>> {
```

Comment:
Note: returning a vec here because we'll have lowerings like:

 ```rust
(foo, bar): (u256, u256) = my_tuple
```

to

```rust
foo: u256 = my_tuple.item0
bar: u256 = my_tuple.item1
```

which requires that we be able to map a single statement to multiple statements.

Reply:
This is a good candidate for `SmallVec<[Node<..>, 1]>` in some possible future performance optimization.

Reply:
Makes sense. I'll track this in an issue.